gapis/api/templates/api.go.tmpl

```diff
@@ -341,12 +342,29 @@ import (
   }
 
   // Clone makes a deep copy of this map.
-  func (m {{$name}}) Clone(ϟa arena.Arena) {{$name}} {
+  func (m {{$name}}) Clone(ϟa arena.Arena, ϟseen ϟapi.CloneContext) {{$name}} {
+    if done, ok := ϟseen[m.Map]; ok {
```

Comment:
Done is probably not the best name here.
`existing` maybe?

Reply:
Done.